File: src/main/java/com/github/bpazy/eql/extractor/BaseMethodCallExtractor.java
package com.github.bpazy.eql.extractor;

import com.intellij.openapi.project.Project;

/**
 * @author ziyuan
 * created on 2018/1/15
 */
public abstract class BaseMethodCallExtractor {
    public abstract boolean filter(String methodName, String argName);

    public abstract Object invoke(Project project, Object arg);
}

File: src/main/java/com/github/bpazy/eql/extractor/BaseMethodCallExtractor.kt
package com.github.bpazy.eql.extractor

import com.intellij.openapi.project.Project

/**
 * @author ziyuan
 * created on 2018/1/15
 */
abstract class BaseMethodCallExtractor {
    abstract fun filter(methodName: String, argName: String?): Boolean

    abstract fun invoke(project: Project, arg: Any): Any
}
